### PR TITLE
Telemetry panel

### PR DESCRIPTION
**Pull request by spryor**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
First pass of the telemetry panel on the dashboard
- add charting library
- get polling working
TODO: Location of calls will be migrated to a single source of truth when all the component data usage is finalized during the dashboard completing process

### CODE REVIEW

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    );
+  }
+
+  setTelemetryKey = telemetryKey => () => this.setState({ telemetryKey });
+
+  render() {
+    const { telemetryKeys, telemetry, telemetryKey } = this.state;
+    return (
+      <Panel>
+        <PanelHeader>Telemetry</PanelHeader>
```
Comment (ppathan) [inline src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: Do we need to localize this text? If yes we should move it to a separate file

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            tooltip: true,
+            yAxisState: 'shared' // Default to all values being on the same axis
+          },
+          chartColors.map(color => ({ color }))
+        );
+      }, 10);
+    }
+  }
+
+  getData() {
```
Comment (ppathan) [inline src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: Can you add a comment what this method is doing? Pretty hard to follow

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+} from 'components/pages/dashboard/panel';
+import { joinClasses } from 'utilities';
+
+import './telemetryPanel.css';
+// TODO: find a way to import without the relative path
+import '../../../../../../node_modules/tsiclient/tsiclient.css';
+
+const telemetryChartId = 'device-telemetry-chart';
+
+const chartColors = [
```
Comment (ppathan) [inline src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: May be add actual color names in comments for clarity

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js (hunk 2 above)
Comment (spryor) [reply src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: Yeah, the calls in this file will be moved to either the redux store or the parent component (`dashboard.js`). I'll document the calls there (and the naming I have in mind will help I think).

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js (hunk 3 above)
Comment (spryor) [reply src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: I don't know how helpful that will actually be. We don't have names from design and even then it will be stuff like 'stone' or 'ash'. These colors will be moved to a config later on so the redux store can switch them out when the theme changes.

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js (hunk 1 above)
Comment (spryor) [reply src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: Yes it does need to be localized. I haven't done so yet since this is the first pass. I'll do the localization pass when the functionality is working correctly. What do you mean by move to a separate file? You mean move the string? When the string is localized it will be moved to the `public/locales/en/translation.js` file.

Comment (ppathan) [reply src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: Yes moving to the locales folder is what i meant. It is ok to do it later

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js (hunk 2 above)
Comment (ppathan) [reply src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: If you are already tracking this then it is fine otherwise open a bug.

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js (hunk 1 above)
Comment (ppathan) [reply src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: If you are already tracking this then it is fine otherwise open a bug.

**Comment by ppathan** [comment]
@spryor Can you please fix the build failures before check-in?

File: src/services/telemetryService.js [hunk 4]
```diff
@@ -25,12 +31,37 @@ export class TelemetryService {
...
       .map(toAlarmsForRuleModel);
   }
+
+  /** Returns a telemetry events */
+  static getTelemetryByMessages(params = {}) {
+    return HttpClient.get(`${ENDPOINT}messages?${stringify(params)}`)
+      .map(toMessagesModel);
+  }
+
+  static getTelemetryByDeviceIdP1M(deviceIds = '') {
```
Comment (avanikp) [inline src/services/telemetryService.js]: What is P1m, PT1M, P15M, PT15M ? Can you add comments for these.

File: src/services/telemetryService.js [hunk 5]
```diff
@@ -25,12 +31,37 @@ export class TelemetryService {
...
+    return TelemetryService.getTelemetryByMessages({
+      from: 'NOW-PT1M',
+      to: 'NOW',
+      order: 'desc',
+      devices: encodeURIComponent(deviceIds)
+    });
+  }
+
+  static getTelemetryByDeviceIdP15M(deviceIds = '') {
```
Comment (avanikp) [inline src/services/telemetryService.js]: Instead of having getTelemetryByDeviceIdP1M and getTelemetryByDeviceIdP15M, can we add a generic method that takes a param for P1M, P15M ?

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+          {
+            grid: false,
+            legend: 'compact',
+            noAnimate, // If the telemetryKey changes, animate
+            tooltip: true,
+            yAxisState: 'shared' // Default to all values being on the same axis
+          },
+          chartColors.map(color => ({ color }))
+        );
+      }, 10);
```
Comment (avanikp) [inline src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: suggest making 10 a const

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    }
+  }
+
+  getData() {
+    this.subscription = TelemetryService.getTelemetryByDeviceIdP15M()
+      .flatMap(response =>
+        Observable.interval(2000)
```
Comment (avanikp) [inline src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: nit: declare 2000 as a const

File: src/services/telemetryService.js (hunk 4 above)
Comment (spryor) [reply src/services/telemetryService.js]: These are time range constants. These will be documented later. The process I'm currently following is get the basic setup working per dashboard panel then I will be doing a unification pass (making use of duplicated calls etc. by moving the calls into the parent component or the redux store). So this is future work.

File: src/services/telemetryService.js (hunk 5 above)
Comment (spryor) [reply src/services/telemetryService.js]: There is a generic method above these `getTelemetryByMessages`. These methods are only temporary though, see comment above.

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js (hunk 7 above)
Comment (spryor) [reply src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: Temporary constant so not adding to the config

File: src/components/pages/dashboard/panels/telemetry/telemetryPanel.js (hunk 6 above)
Comment (spryor) [reply src/components/pages/dashboard/panels/telemetry/telemetryPanel.js]: I'll take this under consideration but this call will later be wrapped in a component to hide the tsiChart config data. Since this isn't strictly speaking an app constant I would lean away from making an app constant.
